Tools/cases_generator/generate_cases.py

```diff
@@ -0,0 +1,134 @@
+"""Generate the main interpreter switch."""
+
+# Write the cases to generated_cases.c.h, which is #included in ceval.c.
+
+# TODO: Reuse C generation framework from deepfreeze.py?
+
+import argparse
+from dataclasses import dataclass
+import re
+import sys
+
+import parser
+
+arg_parser = argparse.ArgumentParser()
+arg_parser.add_argument("-i", "--input", type=str, default="Python/bytecodes.c")
+arg_parser.add_argument("-o", "--output", type=str, default="Python/generated_cases.c.h")
+arg_parser.add_argument("-c", "--compare", action="store_true")
+arg_parser.add_argument("-q", "--quiet", action="store_true")
+
+
+def eopen(filename, mode="r"):
+    if filename == "-":
+        if "r" in mode:
+            return sys.stdin
+        else:
+            return sys.stdout
+    return open(filename, mode)
+
+
+def leading_whitespace(line):
+    return len(line) - len(line.lstrip())
+
+
+@dataclass
+class Instruction:
```

Comment:
Minor: I'm not quite sure what this gives us over `parser.InstDef`?

Reply:
Not much, I'll clean it up.